config/bidderinfo.go

```diff
@@ -177,94 +166,344 @@ type SyncerEndpoint struct {
 	UserMacro string `yaml:"userMacro" mapstructure:"user_macro"`
 }
 
-// Override returns a new SyncerEndpoint object where values in the original are replaced by non-empty/non-default
-// values in the override. No changes are made to the original or override SyncerEndpoint.
-func (s *SyncerEndpoint) Override(original *SyncerEndpoint) *SyncerEndpoint {
-	if s == nil && original == nil {
-		return nil
+// LoadBidderInfo parses all static/bidder-info/{bidder}.yaml files from the file system.
+func LoadBidderInfo(path string) (BidderInfos, error) {
+	bidderConfigs, err := ioutil.ReadDir(path)
+	if err != nil {
+		log.Fatal(err)
 	}
 
-	var copy SyncerEndpoint
-	if original != nil {
-		copy = *original
+	infos := BidderInfos{}
+
+	for _, bidderConfig := range bidderConfigs {
+		if bidderConfig.IsDir() {
+			continue //ignore directories
+		}
+		fileName := bidderConfig.Name()
+		filePath := fmt.Sprintf("%v/%v", path, fileName)
+		data, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			return nil, err
+		}
+
+		info := BidderInfo{}
+		if err := yaml.Unmarshal(data, &info); err != nil {
+			return nil, fmt.Errorf("error parsing yaml for bidder %s: %v", fileName, err)
+		}
+
+		bidderName := strings.Split(fileName, ".yaml")
+		infos[(bidderName[0])] = info
```

Comment:
Is it guaranteed that there will be data present at index 0 here? Just want to check

Reply:
good point. Added check if file is a yaml file and array len.